mflowgen/common/scripts/fix-shorts.tcl

```diff
@@ -0,0 +1,132 @@
+########################################################################
+# Save route blockage info in name() box() and layer() arrays
+set i 0
+foreach ptr [get_db route_blockages] {
+    select_obj $ptr
+
+    set blockage_names($i)  [get_db $ptr .name]
+    set blockage_boxes($i)  [get_db $ptr .rects]
+    set blockage_layers($i) [get_db $ptr .layer.name]
+
+    # echo $i $ptr $name($i)
+    # printf "%2d %s %s\n" $i $ptr $name($i)
+    incr i
+}
+set nblockages $i
+
+########################################################################
+# Display list of blockages found e.g.
+#    0 RouteBlk -layer M3 -box {15.93 1670.88 77.225 1898.352} -name def
+#    1 RouteBlk -layer AP -box {0.0 0.0 0.3 1900.416} -name left
+#    2 RouteBlk -layer M9 -box {0.0 0.0 0.3 1900.416} -name left
+#   ...
+#   52 RouteBlk -layer M3 -box {15.93 765.504 77.225 992.976} -name def
+#   53 RouteBlk -layer M3 -box {15.93 539.136 77.225 766.608} -name def
+#   54 RouteBlk -layer M3 -box {15.93 312.816 77.225 540.288} -name def
+#   55 RouteBlk -layer M3 -box {15.93 86.448 77.225 313.92} -name def
+echo "@file_info: Found and saved info for $nblockages route blockages"
+for { set i 0 } { $i < $nblockages } { incr i } {
+    printf "#   %2d " $i
+    echo RouteBlk \
+        -layer $blockage_layers($i) \
+        -box $blockage_boxes($i)    \
+        -name $blockage_names($i)   \
+        ;
+}
+
+########################################################################
+# Delete route blockages (will restore them again later)
+echo "@file_info: Deleting route blockages"
+deleteRouteBlk -all
+redraw; sleep 1
+
```

Comment:
What's the reason for getting rid of the blockages temporarily? This could cause the router to violate the blockages while trying to fix the shorts... was it unable to fix shorts with the blockages in place?

Reply:
Correct. Blockages existed that prevented the shorts from being fixed. After the blockages are restored, drc is re-checked to verify that we didn't re-route through a forbidden area.